src/pages/index.js

```diff
@@ -0,0 +1,345 @@
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardElement.dataset.id = data._id;
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = `Image of ${data.name}`;
+
+  cardElement.dataset.id = data._id; // ✅ Store card ID in dataset
+
+  // ✅ Ensure `data.likes` exists before checking `.some()`
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  // Attach event listeners correctly
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaption.textContent = data.name;
+    previewModalImage.src = data.link;
+    previewModalImage.alt = `Image of ${data.name}`;
+  });
+
+  return cardElement;
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(
+    editFormElement,
+    [editModalNameInput, editModalDescriptionInput],
+    settings
+  );
+
+  openModal(editModal);
+});
+
+editModalCloseBtn.addEventListener("click", () => {
+  console.log("Closing edit modal...");
+  closeModal(document.querySelector("#edit-modal"));
+});
+
+cardModalBtn.addEventListener("click", () => {
+  openModal(cardModal);
+});
+
+cardModalCloseBtn.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+
+cardPreviewCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+
+cardForm.addEventListener("submit", handleAddCardSubmit);
+
+//Avatar Modal Form Elements
+
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarCloseBtn.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+document.querySelectorAll(".modal__close-btn").forEach((btn) => {
+  btn.addEventListener("click", (evt) => {
+    const modal = evt.target.closest(".modal");
+    if (!modal) return;
+
+    console.log(`Attempting to close modal: ${modal.id}`);
+
+    // ✅ Ensure only `edit-modal` closes
+    if (modal.id === "edit-modal") {
+      closeModal(modal);
+    }
+  });
+});
+
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
+//Delete form elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  deleteForm.addEventListener("submit", handleDeleteSubmit);
```

Comment:
`deleteForm.addEventListener("submit", handleDeleteSubmit);` should be added only **once** outside the opening function, otherwise there will be `memory leak`